chi/lease.py

```diff
@@ -155,192 +160,282 @@ def lease_create_nodetype(*args, **kwargs):
-        manager
-    :param kwargs: Parameters passed through to
-        :py:func:`lease_create_nodetype` and in turn
-        :py:func:`lease_create_args`
+    Attributes:
+        name (str): The name of the lease.
+        start_date (str): The start date of the lease in the format specified by BLAZAR_TIME_FORMAT.
+        end_date (str): The end date of the lease in the format specified by BLAZAR_TIME_FORMAT.
+        id (str): The ID of the lease.
+        status (str): The status of the lease.
+        user_id (str): The ID of the user associated with the lease.
+        project_id (str): The ID of the project associated with the lease.
+        created_at (datetime): The creation date of the lease.
+        node_reservations (list): List of node reservations associated with the lease.
+        fip_reservations (list): List of floating IP reservations associated with the lease.
+        network_reservations (list): List of network reservations associated with the lease.
+        _events (list): List of events associated with the lease.
+
+    Methods:
+        add_node_reservation: Adds a node reservation to the lease.
+        add_fip_reservation: Adds a floating IP reservation to the lease.
+        add_network_reservation: Adds a network reservation to the lease.
+        submit: Submits the lease for creation.
+        wait: Waits for the lease to reach a specific status.
+        refresh: Refreshes the lease data from the Blazar API.
+        delete: Deletes the lease.
+        show: Displays the lease details.
+
+    Properties:
+        events: List of events associated with the lease.
+        status: The status of the lease.
     """
-
-    def __init__(self, **kwargs):
-        kwargs.setdefault("session", session())
-
-        self.session = kwargs.pop("session")
-        self.blazar = blazar(session=self.session)
-        self.neutron = neutron(session=self.session)
-
-        self.lease = None
-
-        self._servers = {}
-
-        self._sequester = kwargs.pop("sequester", False)
-
-        kwargs.setdefault("_preexisting", False)
-        self._preexisting = kwargs.pop("_preexisting")
-
-        kwargs.setdefault("_no_clean", False)
-        self._noclean = kwargs.pop("_no_clean")
-
-        prefix = kwargs.pop("prefix", "")
-        rand = random_base32(6)
-        self.prefix = f"{prefix}-{rand}" if prefix else rand
-
-        kwargs.setdefault("name", self.prefix)
-
-        if self._preexisting:
-            self.id = kwargs["_id"]
-            self.refresh()
+    def __init__(self, name: str,
+                 start_date: datetime = None,
+                 end_date: datetime = None,
+                 duration: timedelta = None,
+                 lease_json: dict = None):
+        self.name = name
+        if start_date:
+            self.start_date = start_date.strftime(BLAZAR_TIME_FORMAT)
+        else:
+            self.start_date = (utcnow() + timedelta(minutes=1)).strftime(BLAZAR_TIME_FORMAT)
```

Comment:
We should actually set this to the string `'now'`, which is a special case in blazar to handle this.